naga-cli/src/bin/naga.rs

```diff
@@ -629,6 +662,53 @@ fn write_output(
     Ok(())
 }
 
+fn bulk_validate(args: Args, params: &Parameters) -> Result<(), Box<dyn std::error::Error>> {
+    let mut invalid = vec![];
+    for input_path in args.files {
+        let path = Path::new(&input_path);
+        let input = fs::read(path)?;
+
+        let Parsed { module, input_text } = match parse_input(path, input, params) {
+            Ok(parsed) => parsed,
+            Err(error) => {
+                invalid.push(input_path.clone());
+                eprintln!("Error validating {}:", input_path);
+                eprintln!("{error}");
```

Comment:
nitpick: I'd combine these into a single lock-'n'-write with `eprintln!("Error validating {}:\n{}", …)`.

Reply:
Fun fact (not intended to dissuade you): you can also use the `concat!(...)` macro, too:

```rust
eprint!(
    concat!(
        "Line 1: {}\n”,
        "Line 2: {}\n",
    ),
    arg1,
    arg2
);
```